src/test/java/dev/openfeature/sdk/providers/memory/InMemoryProviderTest.java

```diff
@@ -0,0 +1,88 @@
+package dev.openfeature.sdk.providers.memory;
+
+import com.google.common.collect.ImmutableMap;
+import dev.openfeature.sdk.Client;
+import dev.openfeature.sdk.OpenFeatureAPI;
+import dev.openfeature.sdk.Value;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static dev.openfeature.sdk.MutableStructure.mapToStructure;
+import static dev.openfeature.sdk.testutils.TestFlagsUtils.buildFlags;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class InMemoryProviderTest {
+
+    private static Client client;
+
+    private static final AtomicBoolean isProviderReady = new AtomicBoolean(false);
+
+    private static final AtomicInteger configurationChangedEventCount = new AtomicInteger(0);
+
+    @SneakyThrows
+    @BeforeAll
+    static void beforeAll() {
+        Map<String, Flag<?>> flags = buildFlags();
+        InMemoryProvider provider = new InMemoryProvider(flags);
+        OpenFeatureAPI.getInstance().setProvider(provider);
+
+        // TODO: setProvider with wait for init, pending https://github.com/open-feature/ofep/pull/80
+        Thread.sleep(500);
+
+        client = OpenFeatureAPI.getInstance().getClient();
+
+        client.onProviderReady(eventDetails -> isProviderReady.set(true));
+
+        client.onProviderConfigurationChanged(eventDetails -> configurationChangedEventCount.incrementAndGet());
```

Comment:
Just curious why you `incrementAndGet` here, and then assert `2` later, instead of just incrementing here and asserting 1 later? Is that testing something that I don't understand?

Reply:
line 43: _updateFlags(flags)_ - update config by replacing all flags
line 44: _updateFlag("addedFlag", Flag.builder()...)_ - update a single flag

each of these method expected to emit ProviderConfigurationChanged event - this provides test coverage including events for each of the 2 methods.